### PR TITLE
Translate thinking-in-react.md

### PR DESCRIPTION
**Pull request by ameetmadan**

### CODE REVIEW

**Comment by ChrisB9** [comment]
Just as a hint: https://github.com/reactjs/de.react.dev/issues/26
Please change the text to 'du'

File: src/content/learn/thinking-in-react.md [hunk 1]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 <img src="/images/docs/s_thinking-in-react_ui.png" width="300" style={{margin: '0 auto'}} />
 
-To implement a UI in React, you will usually follow the same five steps.
+Um eine Benutzeroberfläche in React zu implementieren, folgst du normalerweise denselben fünf Schritten.
 
-## Step 1: Break the UI into a component hierarchy {/*step-1-break-the-ui-into-a-component-hierarchy*/}
+## Schritt 1: Zerlege die Benutzeroberfläche in eine Komponentenhierarchie {/*step-1-break-the-ui-into-a-component-hierarchy*/}
 
-Start by drawing boxes around every component and subcomponent in the mockup and naming them. If you work with a designer, they may have already named these components in their design tool. Ask them!
+Beginne damit, Kästchen um jede Komponente und Unterkomponente im Mockup zu zeichnen und sie zu benennen. Wenn du mit einem Designer zusammenarbeitest, hat er diese Komponenten vielleicht schon in seinem Designtool benannt. Frage ihn oder sie!
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:36]:
```suggestion
Beginne damit, Kästchen um jede Komponente und Unterkomponente im Mockup zu zeichnen und sie zu benennen. Wenn du mit Designern zusammenarbeitest, haben sie diese Komponenten vielleicht schon in ihrem Designtool benannt. Frage sie!
```

File: src/content/learn/thinking-in-react.md [hunk 2]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 
-Depending on your background, you can think about splitting up a design into components in different ways:
+Je nach Ihrem Hintergrund kannst du darüber nachdenken, ein Design auf verschiedene Weise in Komponenten aufzuteilen:
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:38]:
```suggestion
Je nach deinem Hintergrund kannst du darüber nachdenken, ein Design auf verschiedene Weisen in Komponenten aufzuteilen:
```

File: src/content/learn/thinking-in-react.md [hunk 3]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 
-* **Programming**--use the same techniques for deciding if you should create a new function or object. One such technique is the [single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle), that is, a component should ideally only do one thing. If it ends up growing, it should be decomposed into smaller subcomponents. 
-* **CSS**--consider what you would make class selectors for. (However, components are a bit less granular.)
-* **Design**--consider how you would organize the design's layers.
+* **Programmierung** - Verwende dieselben Techniken, um zu entscheiden, ob du eine neue Funktion oder ein neues Objekt erstellen sollten. Eine dieser Techniken ist das [Prinzip der einzigen Verantwortung](https://en.wikipedia.org/wiki/Single_responsibility_principle), d.h. eine Komponente sollte idealerweise nur eine Aufgabe erfüllen. Wenn sie größer wird, sollte sie in kleinere Unterkomponenten zerlegt werden.
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:40]:
```suggestion
* **Programmierung** - Verwende dieselben Techniken, um zu entscheiden, ob du eine neue Funktion oder ein neues Objekt erstellen solltest. Eine dieser Techniken ist das [Single-Responsibility-Prinzip](https://de.wikipedia.org/wiki/Single-Responsibility-Prinzip), d.h. eine Komponente sollte idealerweise nur eine Aufgabe erfüllen. Wenn sie größer wird, sollte sie in kleinere Unterkomponenten zerlegt werden.
```

File: src/content/learn/thinking-in-react.md [hunk 4]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
+* **CSS**--Du solltest überlegen, wofür Klassenselektoren erstellt werden können. (Komponenten sind jedoch etwas weniger granular.)
+* **Design**-überlege, wie die Ebenen des Designs organisiert werden können.
 
-If your JSON is well-structured, you'll often find that it naturally maps to the component structure of your UI. That's because UI and data models often have the same information architecture--that is, the same shape. Separate your UI into components, where each component matches one piece of your data model.
+Wenn Ihr JSON gut strukturiert ist, wirst du oft feststellen, dass es sich auf natürliche Weise der Komponentenstruktur Ihrer Benutzeroberfläche anpasst. Das liegt daran, dass UI- und Datenmodelle oft dieselbe Informationsarchitektur haben, d. h. dieselbe Form. Unterteile die Benutzeroberfläche in Komponenten, wobei jede Komponente einem Teil Ihres Datenmodells entspricht.
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:44]:
```suggestion
Wenn dein JSON gut strukturiert ist, wirst du oft feststellen, dass es sich auf natürliche Weise der Komponentenstruktur deiner Benutzeroberfläche anpasst. Das liegt daran, dass UI- und Datenmodelle oft dieselbe Informationsarchitektur haben, d. h. dieselbe Form. Unterteile die Benutzeroberfläche in Komponenten, wobei jede Komponente einem Teil deines Datenmodells entspricht.
```

File: src/content/learn/thinking-in-react.md [hunk 5]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
+Wenn du dir `ProductTable` (lavender) ansehen, wirst du feststellen, dass der Tabellenkopf (der die Bezeichnungen "Name" und "Preis" enthält) keine eigene Komponente ist. Dies ist eine Frage der Vorliebe, du kannst so oder so vorgehen. In diesem Beispiel ist sie ein Teil von "ProductTable", weil sie innerhalb der Liste von `ProductTable` erscheint. Wenn diese Kopfzeile jedoch zu komplex wird (z.B. wenn du eine Sortierung hinzufügst), kannst du sie in eine eigene Komponente `ProductTableHeader` verschieben.
 
-Now that you've identified the components in the mockup, arrange them into a hierarchy. Components that appear within another component in the mockup should appear as a child in the hierarchy:
+Nachdem du nun die Komponenten im Mockup identifiziert hast, ordnen sie in einer Hierarchie an. Komponenten, die innerhalb einer anderen Komponente im Mockup erscheinen, sollten in der Hierarchie als untergeordnete Komponenten erscheinen:
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:66]:
```suggestion
Nachdem du nun die Komponenten im Mockup identifiziert hast, ordne sie in einer Hierarchie an. Komponenten, die innerhalb einer anderen Komponente im Mockup erscheinen, sollten in der Hierarchie als untergeordnete Komponenten erscheinen:
```

File: src/content/learn/thinking-in-react.md [hunk 6]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 
-## Step 2: Build a static version in React {/*step-2-build-a-static-version-in-react*/}
+## Schritt 2: Erstellen einer statischen Version in React {/*step-2-build-a-static-version-in-react*/}
 
-Now that you have your component hierarchy, it's time to implement your app. The most straightforward approach is to build a version that renders the UI from your data model without adding any interactivity... yet! It's often easier to build the static version first and add interactivity later. Building a static version requires a lot of typing and no thinking, but adding interactivity requires a lot of thinking and not a lot of typing.
+Nun, da du deine Komponentenhierarchie hast, ist es an der Zeit, Ihre Anwendung zu implementieren. Der einfachste Ansatz besteht darin, eine Version zu erstellen, die die Benutzeroberfläche aus Ihrem Datenmodell wiedergibt, ohne Interaktivität hinzuzufügen... noch nicht! Oft ist es einfacher, zuerst die statische Version zu erstellen und die Interaktivität später hinzuzufügen. Die Erstellung einer statischen Version erfordert viel Tipparbeit und kein Nachdenken, aber das Hinzufügen von Interaktivität erfordert viel Nachdenken und nicht viel Tipparbeit.
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:76]:
```suggestion
Nun, da du deine Komponentenhierarchie hast, ist es an der Zeit, deine Anwendung zu implementieren. Der einfachste Ansatz besteht darin, eine Version zu erstellen, die die Benutzeroberfläche aus deinem Datenmodell wiedergibt, ohne Interaktivität hinzuzufügen... noch nicht! Oft ist es einfacher, zuerst die statische Version zu erstellen und die Interaktivität später hinzuzufügen. Die Erstellung einer statischen Version erfordert viel Tipparbeit und kein Nachdenken, aber das Hinzufügen von Interaktivität erfordert viel Nachdenken und nicht viel Tipparbeit.
```

File: src/content/learn/thinking-in-react.md [hunk 7]
```diff
@@ -25,59 +25,59 @@ The JSON API returns some data that looks like this:
...
 
-To build a static version of your app that renders your data model, you'll want to build [components](/learn/your-first-component) that reuse other components and pass data using [props.](/learn/passing-props-to-a-component) Props are a way of passing data from parent to child. (If you're familiar with the concept of [state](/learn/state-a-components-memory), don't use state at all to build this static version. State is reserved only for interactivity, that is, data that changes over time. Since this is a static version of the app, you don't need it.)
+Um eine statische Version Ihrer Anwendung zu erstellen, die Ihr Datenmodell wiedergibt, solltest du [Komponenten](/learn/your-first-component) erstellen, die andere Komponenten wiederverwenden und Daten mithilfe von [props.](/learn/passing-props-to-a-component) weitergeben. (Wenn du mit dem Konzept von [state](/learn/state-a-components-memory) vertraut bist, verwende es für die Erstellung dieser statischen Version überhaupt keinen State. State ist nur für Interaktivität reserviert, d.h. für Daten, die sich mit der Zeit ändern. Da es sich um eine statische Version der Anwendung handelt, brauchst du es nicht).
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:78]:
```suggestion
Um eine statische Version deiner Anwendung zu erstellen, die dein Datenmodell wiedergibt, solltest du [Komponenten](/learn/your-first-component) erstellen, die andere Komponenten wiederverwenden und Daten mithilfe von [props.](/learn/passing-props-to-a-component) weitergeben. (Wenn du mit dem Konzept von [state](/learn/state-a-components-memory) vertraut bist, verwende für die Erstellung dieser statischen Version überhaupt keinen State. State ist nur für Interaktivität reserviert, d.h. für Daten, die sich mit der Zeit ändern. Da es sich um eine statische Version der Anwendung handelt, brauchst du es nicht).
```

File: src/content/learn/thinking-in-react.md [hunk 8]
```diff
@@ -195,93 +195,94 @@ td {
...
+Der State ist die minimale Menge an sich ändernden Daten, die sich deine App merken muss. Das wichtigste Prinzip für die Strukturierung des States ist [DRY (Don't Repeat Yourself)] (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Überlege dir die absolut minimale Darstellung des States, den deine Anwendung braucht, und berechne alles andere nach Bedarf. Wenn du zum Beispiel eine Einkaufsliste erstellst, kannst du die Artikel als Array in state speichern. Wenn du auch die Anzahl der Artikel in der Liste anzeigen willst, speicherst du die Anzahl der Artikel nicht als weiteren Statewert, sondern liest die Länge deines Arrays.
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:212]:
```suggestion
Der State ist die minimale Menge an sich ändernden Daten, die sich deine App merken muss. Das wichtigste Prinzip für die Strukturierung des States ist [DRY (Don't Repeat Yourself)] (https://de.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself). Überlege dir die absolut minimale Darstellung des States, den deine Anwendung braucht, und berechne alles andere nach Bedarf. Wenn du zum Beispiel eine Einkaufsliste erstellst, kannst du die Artikel als Array in state speichern. Wenn du auch die Anzahl der Artikel in der Liste anzeigen willst, speicherst du die Anzahl der Artikel nicht als weiteren Statewert, sondern liest die Länge deines Arrays.
```

File: src/content/learn/thinking-in-react.md [hunk 9]
```diff
@@ -195,93 +195,94 @@ td {
...
-2. The search text the user has entered
-3. The value of the checkbox
-4. The filtered list of products
+1. Die ursprüngliche Liste der Produkte
+2. Der Suchtext, den der Benutzer eingegeben hat
+3. Der Wert des Kontrollkästchens
+4. Die gefilterte Liste der Produkte
 
-Which of these are state? Identify the ones that are not:
+Welche davon sind State? Identifiziere die Produkte, die es nicht sind:
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:221]:
```suggestion
Welche davon sind State? Identifiziere die, die es nicht sind:
```

File: src/content/learn/thinking-in-react.md [hunk 10]
```diff
@@ -195,93 +195,94 @@ td {
...
 
-* Does it **remain unchanged** over time? If so, it isn't state.
-* Is it **passed in from a parent** via props? If so, it isn't state.
-* **Can you compute it** based on existing state or props in your component? If so, it *definitely* isn't state!
+* Bleiben sie im Laufe der Zeit **unverändert**? Wenn ja, handelt es sich nicht um einen State.
+* Wird es **von einem Parent** über Props übergeben? Wenn ja, handelt es sich nicht um einen State.
+* **Kannst du ihn** auf der Grundlage von bestehenden States oder Props in deiner Komponente berechnen? Wenn ja, handelt es sich *definitiv* nicht um einen State!
```
Comment (ChrisB9) [inline src/content/learn/thinking-in-react.md:225]:
```suggestion
* **Berechnest du ihn** auf der Grundlage von bestehenden States oder Props in deiner Komponente? Wenn ja, handelt es sich *definitiv* nicht um einen State!
```

**Review by ChrisB9 — COMMENTED** [review]
I went through some of the changes already and got some suggestions

**Comment by ameetmadan** [comment]
Accepted your changes, did miss some of the things which should have been obvious. Thanks for reviewing!

**Comment by ameetmadan** [comment]
Is there anything left from my side to do?

**Review by ChrisB9 — APPROVED** [review]

**Comment by ChrisB9** [comment]
@ameetmadan thanks for the reminder, no everything looked good as far as i could say.

**Pull request merged**